.github/scripts/check_skip_ci.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+set -euo pipefail
+
+# Get the list of changed files
+# Using `git merge-base` ensures that we're always comparing against the correct branch point.
+#For example, given the commits:
+#
+# A---B---C---D---W---X---Y---Z # origin/main
+#             \---E---F         # feature/branch
+#
+# ... `git merge-base origin/$SKIP_CHECK_BRANCH HEAD` would return commit `D`
+# `...HEAD` specifies from the common ancestor to the latest commit on the current branch (HEAD)..
+files_to_check=$(git diff --name-only "$(git merge-base origin/$SKIP_CHECK_BRANCH HEAD~)"...HEAD)
+
+# Define the directories to check
+skipped_directories=("_doc/" ".changelog/")
+
+# Loop through the changed files and find directories/files outside the skipped ones
+files_to_check_array=($files_to_check)
```

Comment:
We were not converting to an array in the original script - it happened to still work due to multiline matches, but this allows us to print individual debug logs more accurately